### DISCUSSION TITLE
Is there a way to compress content from an strIO object?

### DISCUSSION THREAD

**Discussion by ValerioParisMitritsakis** [miurahr/py7zr#569] [Q&A]
I am creating a file held by an strIO object which I would then like to compress via py7zr alas it will only accept actual files and direcories and not strIO objects. Is there any chance you could add this feature?

**Comment by miurahr** [comment]
Because 7zip is binary compression tool, string compression by `writestr` is a just a helper method to encode with UTF-8.

You can write  wrapper function to loop reading line or string from `StrIO` object and calling `writestr`.

**Comment by ValerioParisMitritsakis** [comment]
Thank you for taking the time to respond. Would that work with any type of file? Sometimes I am holding excel files (.xlsx format) on strIO objects which I would like to compress (and email) without actually saving their contents on disk.

**Comment by miurahr** [comment]
I don't know how to hold excel binary data into strIO object, so I don't know a answer.
